src/runtime_src/core/tools/xbtracer/src/lib/capture.cpp

```diff
@@ -0,0 +1,617 @@
+  handle = dlopen(lib_name, RTLD_LAZY);
+  if (!handle)
+  {
+    std::lock_guard<std::mutex> lock(dlerror_mutex);
+    // NOLINTNEXTLINE(concurrency-mt-unsafe) - protected by dlerror_mutex
+    std::cerr << "Error loading shared library: " << dlerror() << std::endl;
+    return 0;
+  }
+
+  /**
+   * Get Function address's which are of interest and ignore others.
+   */
+  for (auto& it : func_mangled)
+  {
+    auto ptr_itr = fname2fptr_map.find(it.first);
+
+    if (ptr_itr != fname2fptr_map.end())
+    {
+      void** temp = ptr_itr->second;
+      /* update the original function address in the dispatch table */
+      *temp = (dlsym(handle, it.second.c_str()));
+      if (nullptr == temp)
+        std::cout << "Null Func address received " << std::endl;
+      else
+        fptr2fname_map[*temp] = std::regex_replace(it.first,
+                                      std::regex(R"(\)\s*const)"), ")");
+    }
+    else
+    {
+      // std::cout << "func :: \"" << it.first << "\" not found in"
+      //          << "fname2fptr_map\n";
+    }
+  }
+  return 1;
+}
+
+/**
+ * This function will read mangled API's from library and perform
+ * Demangling operation.
+ */
+int router::load_symbols()
+{
+  constexpr unsigned int symbol_len = 1024;
+
+  path = find_library_path();
+
+  // Open the ELF file
+  std::ifstream elf_file(path, std::ios::binary);
+  if (!elf_file.is_open())
+  {
+    std::cerr << "Failed to open ELF file: " << path << std::endl;
+    return 0;
+  }
+
+  // Read the ELF header
+  Elf64_Ehdr elf_header;
+  std::array<char, sizeof(Elf64_Ehdr)> buffer{0};
+  elf_file.read(buffer.data(), buffer.size());
+  memcpy(&elf_header, buffer.data(), buffer.size());
+
+  // Check ELF magic number
+  if (memcmp(static_cast<const void*>(elf_header.e_ident),
+              static_cast<const void*>(ELFMAG), SELFMAG) != 0)
+  {
+    std::cerr << "Not an ELF file" << std::endl;
+    return 0;
+  }
+
+  // Get the section header table
+  elf_file.seekg((std::streamoff)elf_header.e_shoff);
+  std::vector<Elf64_Shdr> section_headers(elf_header.e_shnum);
+  /* The reinterpret cast is necessary for performing low-level binary file I/O
+   * operations involving a specific memory layout.
+   * The memory layout and size of section_headers match the binary data being
+   * read, ensuring safety and correctness.
+   */
+  // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
+  elf_file.read(reinterpret_cast<char*>(section_headers.data()),
+                (std::streamsize)(elf_header.e_shnum * sizeof(Elf64_Shdr)));
+  if (!elf_file)
+  {
+    std::cerr << "Failed to read section header table" << std::endl;
+    return 0;
+  }
+
+  // Find the symbol table section
+  Elf64_Shdr* symtab_section = nullptr;
+  for (int i = 0; i < elf_header.e_shnum; ++i)
+  {
+    if (section_headers[i].sh_type == SHT_DYNSYM)
+    {
+      symtab_section = &section_headers[i];
+      break;
+    }
+  }
+
+  if (symtab_section == nullptr)
```

Comment:
No common to compare against nullptr.
```suggestion
  if (!symtab_section)
```

Reply:
Done